sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java

```diff
@@ -123,27 +124,28 @@ public EventHubsProducerFactory defaultEventHubsNamespaceProducerFactory(
         /**
          * Creates an Event Hubs message converter.
          *
+         * @param objectMapper A default/customize ObjectMapper
          * @return An Event Hubs message converter.
          */
         @Bean
         @ConditionalOnMissingBean
-        public EventHubsMessageConverter eventHubsMessageConverter() {
-            return new EventHubsMessageConverter();
+        public EventHubsMessageConverter eventHubsMessageConverter(ObjectMapper objectMapper) {
+            return new EventHubsMessageConverter(objectMapper);
         }
 
         /**
          * Creates an Event Hubs template.
          *
          * @param producerFactory An Event Hubs producer factory.
-         * @param messageConverter An Event Hubs message converter.
+         * @param messageConverter An Event Hubs objectProvider message converter.
          * @return An Event Hubs template.
          */
         @Bean
         @ConditionalOnMissingBean
         public EventHubsTemplate eventHubsTemplate(EventHubsProducerFactory producerFactory,
-                                                   EventHubsMessageConverter messageConverter) {
+                                                   ObjectProvider<EventHubsMessageConverter> messageConverter) {
```

Comment:
nit: now that we removed the condition of ObjectMapper class for message converter bean, I suppose it is not possible that there is no message converter bean in the context, thus it's not required to use the ObjectProvider any more.